src/compas/geometry/primitives/frame.py

```diff
@@ -648,187 +649,107 @@ def euler_angles(self, static=True, axes='xyz'):
 
         Parameters
         ----------
-        vector: :obj:`list` of :obj:`float` or :class:`Vector`
-            A vector in local coordinates.
+        object_in_lcs : :class:`Point` or :class:`Vector` or :class:`Frame` or list of float
+            An object in local coordinate system of the frame.
 
         Returns
         -------
-        :class:`Vector`
-            A vector in the world coordinate system.
-
-        Examples
-        --------
-        >>> from compas.geometry import Frame
-        >>> f = Frame([1, 1, 1], [0.68, 0.68, 0.27], [-0.67, 0.73, -0.15])
-        >>> pw1 = [2, 2, 2]
-        >>> pf = f.represent_vector_in_local_coordinates(pw1)
-        >>> pw2 = f.represent_vector_in_global_coordinates(pf)
-        >>> allclose(pw1, pw2)
-        True
+        :class:`Point` or :class:`Vector` or :class:`Frame`
+            The object in the world coordinate frame.
 
-        """
-        T = matrix_from_frame(self)
-        vec = Vector(*vector)
-        vec.transform(T)
-        return vec
-
-    def represent_frame_in_local_coordinates(self, frame):
-        """Represents another frame in the frame's local coordinate system.
-
-        Parameters
-        ----------
-        frame: :class:`Frame`
-            A frame in the world coordinate system.
-
-        Returns
-        -------
-        :class:`Frame`
-            A frame in the frame's local coordinate system.
+        Notes
+        -----
+        If you pass a list of float, it is assumed to represent a point.
 
         Examples
         --------
-        >>> f = Frame([1, 1, 1], [0.68, 0.68, 0.27], [-0.67, 0.73, -0.15])
-        >>> pw1 = Frame([1, 1, 1], [0.707, 0.707, 0], [-0.707, 0.707, 0])
-        >>> pf = f.represent_frame_in_local_coordinates(pw1)
-        >>> pw2 = f.represent_frame_in_global_coordinates(pf)
-        >>> allclose(pw1.point, pw2.point)
-        True
-        >>> allclose(pw1.xaxis, pw2.xaxis)
-        True
-        >>> allclose(pw1.yaxis, pw2.yaxis)
-        True
-
+        >>> from compas.geometry import Frame
+        >>> frame = Frame([1, 1, 1], [0.68, 0.68, 0.27], [-0.67, 0.73, -0.15])
+        >>> pl = Point(1.632, -0.090, 0.573) # point in frame
+        >>> pw = frame.global_coords(pl) # point in wcf
+        >>> frame.local_coords(pw)
+        Point(1.632, -0.090, 0.573)
         """
-        T = Transformation.from_frame(self).inverse()
-        f = frame.copy()
-        f.transform(T)
-        return f
+        T = Transformation.change_basis(self, Frame.worldXY())
+        if isinstance(object_in_lcs, list):
+            return Point(*object_in_lcs).transformed(T)
+        else:
+            return object_in_lcs.transformed(T)
 
-    def represent_frame_in_global_coordinates(self, frame):
-        """Represents another frame in the local coordinate system in the world
-            coordinate system.
+    @staticmethod
+    def local_to_local_coords(frame1, frame2, object_in_frame1):
```

Comment:
same as with `global` (and also applies everywhere) should `local` not be `frame`?

Reply:
Agree to the `global` to `world`. But with I am not sure with the `frame` to `local`.
While it works well with the function-based version (`world_to_frame_coords(frame, point)` or `world_to_frame(frame, point)`), the oo-version does not look so nice: `frame.to_frame(point)` or `frame.to_frame_coords(point)`. There I would prefer keeping the `frame.local_coords(point)`. 

Some options:

| oo | function |
| --- | --- |
| `frame.local_coords(point)`, `frame.world_coords(point)`, `frame.to_local(point)`, `frame.to_world(point)`, | `world_to_local(frame, point)`, `world_to_local_coords(frame, point)`, `local_to_world_coords(frame, point)`, `local_to_world(frame, point)`  |
| `frame.to_frame(point)`, `frame.to_world(point)`, `frame.to_frame_coords(point)`, `frame.to_world_coords(point)` | `world_to_frame(frame, point)`, `world_to_frame_coords(frame, point)`, `frame_to_world_coords(frame, point)`, `frame_to_world(frame, point)` |
| `Frame.local_to_local_coords(f1, f2, point)`,  `Frame.frame_to_frame_coords(f1, f2, point)` | `local_to_local_coords(f1, f2, point)`, `frame_to_frame_coords(f1, f2, point)` |

more opinons, ideas, .. are welcome ;)

Reply:
I prefer `local` much more than `frame` (frame is very ambiguous imo).
I like the option with `to_` prefixes.

Reply:
alright, local it is. 
`to_` versions +1

Reply:
ok, thank you! implemented changes